blog/a-bloop-tour-for-metals-users.md

```diff
@@ -0,0 +1,142 @@
+and take a look at everything in there. Once this information is gathered,
+Metals tells Bloop to compile your project.
+
+At this point, if there are any errors during compilation, diagnostics are
+forwarded from Bloop to Metals, to your client for you to see. If you then fix
+the diagnostic, hit save, the save event is sent to Metals, which then forwards
+that to Bloop to compile what has been changed. You can start to imagine the
+flow from the diagram above.
+
+At this point, one of the questions you may have is how does Bloop start? Does
+Metals start it? Does Metals install Bloop? There can only be one Bloop server
+running on a machine. So when Metals is about to start, one of two things
+happens. Either Metals detects that a Bloop server is running and connects to
+it, or it will start one.
+
+If you've even used the debugging features in Metals, you're also utilizing the
+DAP (Debug Adaptor
+Protocol)[https://microsoft.github.io/debug-adapter-protocol/] support that
+Bloop offers. You can find the entire [debugging reference
+here](https://scalacenter.github.io/bloop/docs/debugging-reference).
+
+## [Bloop CLI](#bloop-cli)
+
+Especially for Metals users, I highly recommend using Bloop CLI. Up above we
+mentioned that compilations are cached for different clients. The power in this
+can be witness when you're in Metals, have you project compiled, and you can run
+a test via the Bloop CLI and see it start basically immediately without another
+compilation happening. Personally, I've fully moved from running test through
+SBT to running them through Bloop CLI simply because of how much faster it is
+for me to go from executing the `bloop test <project>` to seeing the test run.
+If you run the same test through SBT you normally have both a longer startup
+time and also compilation that needs to happen. You can find detailed usage on
+the Bloop CLI [here on the
+website](https://scalacenter.github.io/bloop/docs/cli/tutorial). All of the
+things that you would imagine using like, targeting a specific test suite,
+watching a test suite, passing in arguments, or testing upstream projects all
+exist. You can even jump into a console session in a project using `bloop
+console`.
+
+## [Takeaways and Things to keep in mind](#takeaways-and-things-to-keep-in-mind)
+
+- Whenever you run, test, or debug in Metals, its all being powered by Bloop.
+- Compilations are cached. Consider using the Bloop CLI rather than your build
+    tool to run your tests. You can actually witness this _compile
+    deduplication_ if you are watching a test with `-w` and also compiling a
+    project with Metals. You'll see a message like this in your bloop output:
+
+```shell
+Deduplicating compilation of root from bsp client 'Metals 0.9.0+139-c169b4ce-SNAPSHOT' (since 39.439s)
+```
+
+- There are _many_ integrations with Bloop ranging from [SBT](
+    https://www.scala-sbt.org/) to [Mill](https://github.com/lihaoyi/mill) to
```

Comment:
Mention Gradle and Maven? I don't know how many users run seed or fury 🤔

Reply:
That was sort of intentional. I wanted to emphasize that there are integrations with the most common build tools like SBT and Mill, and also more obscure ones like Seed and Fury. Let me think if there is a better way to re-word this.